src/Common/BackgroundShellCommandHolder.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <memory>
+#include <mutex>
+
+#include <Common/ShellCommand.h>
+#include <unordered_map>
+
+
+namespace DB
+{
+
+/** The holder class for running background shell processes.
+*/
+class BackgroundShellCommandHolder final
+{
+public:
+    void removeCommand(pid_t pid);
+    void addCommand(std::unique_ptr<ShellCommand> command);
+
+private:
+    using ActiveShellCommandsCollection = std::unordered_map<pid_t, std::unique_ptr<ShellCommand>>;
+
+    std::mutex mutex;
+    ActiveShellCommandsCollection active_shell_commands TSA_GUARDED_BY(mutex);
```

Comment:
```suggestion
    ShellCommands shell_commands TSA_GUARDED_BY(mutex);
```

Reply:
I would propose to shorten the name, because we don't have non-active alternatives in the project now.